tests/Unit/Evolution/Systems/NewtonianEuler/BoundaryCorrections/Hll.py

```diff
@@ -0,0 +1,208 @@
+        polytropic_exponent = 2.0
+        sound_speed = np.sqrt(polytropic_constant * polytropic_exponent *
+                              pow(mass_density, polytropic_exponent - 1.0))
+    else:
+        adiabatic_index = 1.3
+        chi = specific_internal_energy * (adiabatic_index - 1.0)
+        kappa_times_p_over_rho_squared = ((adiabatic_index - 1.0)**2 *
+                                          specific_internal_energy)
+        sound_speed = np.sqrt(chi + kappa_times_p_over_rho_squared)
+
+    if normal_dot_mesh_velocity is None:
+        return velocity_dot_normal - sound_speed
+    else:
+        return velocity_dot_normal - sound_speed - normal_dot_mesh_velocity
+
+
+def dg_boundary_terms_mass_density(
+    interior_mass_density, interior_momentum_density, interior_energy_density,
+    interior_normal_dot_flux_mass_density,
+    interior_normal_dot_flux_momentum_density,
+    interior_normal_dot_flux_energy_density,
+    interior_largest_outgoing_char_speed, interior_largest_ingoing_char_speed,
+    exterior_mass_density, exterior_momentum_density, exterior_energy_density,
+    exterior_normal_dot_flux_mass_density,
+    exterior_normal_dot_flux_momentum_density,
+    exterior_normal_dot_flux_energy_density,
+    exterior_largest_outgoing_char_speed, exterior_largest_ingoing_char_speed,
+    use_strong_form):
+
+    lambda_max = np.amax([
+        0., interior_largest_outgoing_char_speed,
+        -exterior_largest_ingoing_char_speed
+    ])
+    lambda_min = np.amin([
+        0., interior_largest_ingoing_char_speed,
+        -exterior_largest_outgoing_char_speed
+    ])
+
+    if use_strong_form:
+        return (lambda_min * (interior_normal_dot_flux_mass_density +
+                              exterior_normal_dot_flux_mass_density) +
+                lambda_max * lambda_min *
+                (exterior_mass_density - interior_mass_density)) / (
+                    lambda_max - lambda_min)
+    else:
+        return (
+            (lambda_max * interior_normal_dot_flux_mass_density + lambda_min *
+             exterior_normal_dot_flux_mass_density) + lambda_max * lambda_min *
+            (exterior_mass_density - interior_mass_density)) / (lambda_max -
+                                                                lambda_min)
+
+
+def dg_boundary_terms_momentum_density(
+    interior_mass_density, interior_momentum_density, interior_energy_density,
+    interior_normal_dot_flux_mass_density,
+    interior_normal_dot_flux_momentum_density,
+    interior_normal_dot_flux_energy_density,
+    interior_largest_outgoing_char_speed, interior_largest_ingoing_char_speed,
+    exterior_mass_density, exterior_momentum_density, exterior_energy_density,
+    exterior_normal_dot_flux_mass_density,
+    exterior_normal_dot_flux_momentum_density,
+    exterior_normal_dot_flux_energy_density,
+    exterior_largest_outgoing_char_speed, exterior_largest_ingoing_char_speed,
+    use_strong_form):
+
+    lambda_max = np.amax([
```

Comment:
@nilsdeppe I got a small question here -- are the arguments in these python functions (e.g. `interior_mass_density`) supposed to be a single number or array? (I haven't got the full sense of how `Test_##.cpp` files are calling python scripts) Because if the latter is true, I should have not used `numpy.amax` here.

Reply:
I think the char speeds in this case are single element arrays. I ended up using `np.maximum` in these types of calls. you can also double check the type using `print(type(interior_largest_outgoing_char_speed))`, and then running either as `./bin/RunTests Unit.NewtonianEuler.Hll` or `ctest --verbose -R Unit.NewtonianEuler.Hll`, then looking for the printed out type.

Reply:
It turned out to be 0-dimensional `numpy.ndarray`, but still the implementation seems to work fine. Do you think it will be safer to add `[()]` to grab out the value? e.g. `interior_largest_outgoing_char_speed[()]`

Reply:
can you use the exact same thing that the rusanov solver does? It seems the most desirable thing would be to use the same code in different cases to avoid people being confused about the different implementations

Reply:
Rusanov solver is using `np.maximum` which works for two arguments, but we have three entries for picking out max (min) value here, so I thought `np.amax` might be simpler than using nested `np.maximum`. On my second thoughts, just using `np.max` and `np.min` instead of `np.amax` and `np.amin` seems less confusing. What do you think?

Reply:
why not just a nested `np.maximum` call? the max and amax are the same function, and they return the maximum value, we want the max element-wise over two arrays, so max and amax are both incorrect

Reply:
Yep. that would be more consistent. Now I got it.